File: python/promptspec/models/prompt_collection.py
"""
Root container for all prompt specifications.
"""
from typing import List
from pydantic import BaseModel, Field
from .prompt_specification import PromptSpecification


class PromptCollection(BaseModel):
    """Root container for all prompt specifications loaded from YAML."""
    
    prompts: List[PromptSpecification] = Field(
        default_factory=list, 
        description="Collection of prompt specifications"
    )

File: src/Models/PromptCollection.cs
using YamlDotNet.Serialization;

namespace PromptSpec.Models
{

    /// <summary>
    /// Root container for all prompt specifications loaded from YAML.
    /// </summary>
    public class PromptCollection
    {
        /// <summary>
        /// Collection of prompt specifications.
        /// </summary>
        [YamlMember(Alias = "prompts")]
        public List<PromptSpecification> Prompts { get; set; }
    }
}
